sdk/src/main/java/com/kin/ecosystem/transfer/presenter/AccountInfoPresenter.java

```diff
@@ -0,0 +1,146 @@
+
+    private AccountInfoManager accountInfoManager;
+    private AccountInfoAsyncTask asyncTask;
+    private boolean taskComplete = false;
+    private boolean completeWithSuceess = false;
+    private boolean isPaused = false;
+
+    public AccountInfoPresenter(@NonNull AccountInfoManager accountInfoManager, @NonNull IAccountInfoView view, @NonNull Intent intent) {
+        this.accountInfoManager = accountInfoManager;
+        processIntent(view, intent);
+        view.attachPresenter(this);
+        startInit();
+    }
+
+    private void processIntent(IAccountInfoView view, Intent intent) {
+        if (intent != null && intent.hasExtra(EXTRA_SOURCE_APP_NAME)) {
+            String sourceApp = intent.getStringExtra(EXTRA_SOURCE_APP_NAME);
+            if (!sourceApp.isEmpty()) {
+                view.updateSourceApp(sourceApp);
+            } else {
+                onError(view);
+            }
+        } else {
+            onError(view);
+        }
+    }
+
+    @Override
+    public void agreeClicked() {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondOk();
+            if (view != null) {
+                view.close();
+            }
+        }
+    }
+
+    @Override
+    public void backButtonPressed() {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondCancel();
+        }
+    }
+
+    @Override
+    public void closeClicked() {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondCancel();
+            if (view != null) {
+                view.close();
+            }
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        asyncTask.cancel(true);
+        asyncTask = null;
+        super.onDetach();
+    }
+
+    @Override
+    public void onResume() {
+        isPaused = false;
+        if (taskComplete) {
+            onInitComplete(completeWithSuceess);
+        }
+    }
+
+    @Override
+    public void onPause() {
+        isPaused = true;
+    }
+
+    private void startInit() {
+        asyncTask = new AccountInfoAsyncTask();
+        asyncTask.execute();
+    }
+
+    private void onError(IAccountInfoView view) {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondError();
+            if (view != null) {
+                view.close();
+            }
+        }
+    }
+
+    private void onTaskComplete(Boolean success) {
+        if (!isPaused) {
+            onInitComplete(success);
+        } else {
+            taskComplete = true;
+            completeWithSuceess = success;
+        }
+    }
+
+    private void onInitComplete(Boolean success) {
+        if (success) {
+            if (view != null) {
+                view.enabledAgreeButton();
+            }
+        } else {
+            onError(getView());
+        }
+    }
+
+    private class AccountInfoAsyncTask extends AsyncTask<Void, Void, Boolean> {
```

Comment:
AccountInfoAsyncTask should be in the Activity as it Android code. the presenter can implement IAsyncTask interface of doInBackground, onPostExecute which will be called from AccountInfoAsyncTask in the Activty

Reply:
we discussed to keep it as is for this case